### PR TITLE
Change python environment and code to python3 for zilliqa repo

### PR DESCRIPTION
**Pull request by chetan-zilliqa**
## Description
Change Python from python 2.7 to python 3.5 inside ubuntu16 docker image and convert zilliqa scripts to python3 format.

## Backward Compatibility

- [ ] This is not a breaking change
- [ ] This is a breaking change

### Implementation

- [ ] **ready for review**

### Integration Test (Core Team)

- [ ] local machine test

- [x] small-scale cloud test

### CODE REVIEW

File: scripts/list_shards.py [hunk 1]
```diff
@@ -63,17 +63,17 @@ def scan_file(fileName):
...
-				print outputstr
+				print(outputstr)
 				for peerIndex in range(len(shardsAndShardPeers[0])):
 					outputstr = "PEER " + str(peerIndex) + ","
 					for shardIndex in range(len(shardsAndShardPeers)):
 						outputstr = outputstr + shardsAndShardPeers[shardIndex][peerIndex] + ","
-					print outputstr
-				print ""
+					print(outputstr)
+				print("")
```
Comment (Revolution1) [inline scripts/list_shards.py:76]:
```
print() # no need to pass an argument to print
```

File: scripts/profile_time.py [hunk 2]
```diff
@@ -166,13 +166,13 @@ def check_miblock_broadcast(line):
 		strSendTime = get_time(line)
 		sendTime = convert_time_string(strSendTime)
 		# Need to get the earliest one to send out
-		if ((not MIBlockSendTime.has_key(blockNumber)) or sendTime < convert_time_string(MIBlockSendTime[blockNumber])):
+		if ((blockNumber not in MIBlockSendTime) or sendTime < convert_time_string(MIBlockSendTime[blockNumber])):
```
Comment (Revolution1) [inline scripts/profile_time.py:169]: no need to parenthesized if statement

File: scripts/report_bastion.py [hunk 3]
```diff
@@ -76,7 +76,7 @@ def generateReport(reportname, lookups, dsNodesSubset, webhookURL):
 	# 24 = Latest VC Tx
 
 	lookupData = dict()
-	for key, val in RESULT_LOOKUP.items():
+	for key, val in list(RESULT_LOOKUP.items()):
```
Comment (Revolution1) [inline scripts/report_bastion.py:79]: no need to list() here

File: scripts/report_bastion.py [hunk 4]
```diff
@@ -185,7 +185,7 @@ def generateReport(reportname, lookups, dsNodesSubset, webhookURL):
 	# 9 = PoW
 
 	dsNodeData = dict()
-	for key, val in RESULT_DSNODE.items():
+	for key, val in list(RESULT_DSNODE.items()):
```
Comment (Revolution1) [inline scripts/report_bastion.py:188]: no need to list() here

File: scripts/scan_consolidate.py [hunk 5]
```diff
@@ -140,7 +140,7 @@ def filter(string, substr):
 
 def get_filenames_for_dir(path, substr):
     all_files = os.listdir(path)
-    return filter(all_files,[substr])
```
Comment (Revolution1) [inline scripts/scan_consolidate.py:143]: filter overrides the built-in filter()

File: scripts/scan_consolidate.py [hunk 6]
```diff
@@ -266,9 +266,9 @@ def search_ds(txpkt_stats, lookup_ident, txpool_stats):
                 else:
                     txpkt_stats[epoch_num][shard_id][lookup_pkt] = \
                         [0, \
-                        max((vals[1] - vals[0]) for fileName, vals in txpkt_stats[epoch_num][shard_id][lookup_pkt].items()), \
+                        max((vals[1] - vals[0]) for fileName, vals in list(txpkt_stats[epoch_num][shard_id][lookup_pkt].items())), \
```
Comment (Revolution1) [inline scripts/scan_consolidate.py:269]: no need to list() here

File: scripts/scan_consolidate.py [hunk 7]
```diff
@@ -459,13 +459,13 @@ def add_rows_for_epoch(epoch_num, lookup_packets, a, b, c, d, e, f, num_shards):
...
-                        max(k for k, v in lookup_packets.items()), \
-                        max(k for k, v in txpkt_stats.items()), \
-                        max(k for k, v in txpool_stats.items()))
+        num_epochs = max(max(k for k, v in list(mb_time_infos.items())), \
+                        max(k for k, v in list(ds_consensus_times.items())), \
+                        max(k for k, v in list(ds_wait_times.items())), \
+                        max(k for k, v in list(lookup_info.items())), \
+                        max(k for k, v in list(lookup_packets.items())), \
+                        max(k for k, v in list(txpkt_stats.items())), \
+                        max(k for k, v in list(txpool_stats.items())))
```
Comment (Revolution1) [inline scripts/scan_consolidate.py:468]:
```
num_epochs = max(itertools.chain(
    mb_time_infos, 
    ds_consensus_times, 
    ...
))
```

File: scripts/stall_checker.py [hunk 8]
```diff
@@ -62,7 +62,7 @@ def generateReport(reportname, lookups, webhookURL, stalInMins):
 
 	lookupData = dict()
 	thetimenow = datetime.datetime.now()
-	for key, val in RESULT_LOOKUP.items():
+	for key, val in list(RESULT_LOOKUP.items()):
```
Comment (Revolution1) [inline scripts/stall_checker.py:65]: no need to list() here

File: docker/Dockerfile.k8s [hunk 9]
```diff
@@ -15,7 +15,8 @@
 #
 # This Dockerfile is incomplete and should only be used with other Dockerfiles
 
-COPY requirements2.k8s.txt ./
+
+COPY requirements3.k8s.txt ./
```
Comment (joel-zilliqa) [inline docker/Dockerfile.k8s:19]: remove this line, we install requirements3.txt on the base image for clarity

File: docker/Dockerfile.k8s [hunk 10]
```diff
@@ -27,5 +28,5 @@ RUN apt-get update && apt-get install -y --no-install-recommends \
     trickle \
     vim \
     && rm -rf /var/lib/apt/lists/* \
-    && pip install -r requirements2.k8s.txt
+    && pip3 install -r requirements3.k8s.txt
```
Comment (joel-zilliqa) [inline docker/Dockerfile.k8s:31]: remove this line too

File: docker/requirements3.k8s.txt [hunk 11]
```diff
@@ -0,0 +1,5 @@
+urllib3==1.25.9
```
Comment (joel-zilliqa) [inline docker/requirements3.k8s.txt:1]: rename this file to requirements3.txt

**Pull request merged**